### PR TITLE
fix: no rubric validation added

### PR DESCRIPTION
**Pull request by Inferato**
**TL;DR -** 
Save ORA w/o prompt cause an error

**What changed?**

- Added no prompt validation
- `validate_rubric` placed on top of validations 

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled
- [ ] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Inferato!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by itsjeyd** [comment]
@Inferato Thank you for your contribution! Please let us know when it is ready for review.

**Comment by Inferato** [comment]
@itsjeyd Now it's ready for review.

**Comment by itsjeyd** [comment]
@Inferato OK great!

@mattcarter This is ready for engineering review by Aurora.

**Comment by itsjeyd** [comment]
@pomegranited Perhaps you could have a look at this PR, [too](https://github.com/openedx/edx-ora2/pull/2101#issuecomment-1924038395)?

**Review by pomegranited — CHANGES_REQUESTED** [review]
Hi @Inferato , thank you for submitting this change! It works as described.

I've tested it and it works great, but have left a note about the translation changes. Once you address these issues, I'll approve and get this merged.

- [x] I tested this on my devstack:
   * Had to enable the `course_live.enable_course_live` waffle flag -- editing ORA components in Studio throws errors otherwise.
   * Added a new ORA2 component to my test course
   * Removed all "prompts" from the component and saved.
     Noted new error: "You must provide at least one prompt."
     (ora2 master raises an IndexError in the backend, but shows the user no details.)
- [x] I read through the code
- [x] I checked for accessibility issues by using my keyboard only during testing
- [x] <strike>Includes documentation</strike> N/A
- [x] User-facing strings are extracted for translation

**Comment by dyudyunov** [comment]
Hi @pomegranited 

Sorry for the delays, Taras is no longer a member of RG.
Currently, I'm gathering all the PRs he worked on. I will check their statuses and update them if necessary.

I will leave the status update comments on each PR's comments ASAP

**Comment by pomegranited** [comment]
Hi @dyudyunov no worries, thank you for letting us know.
If you end up having to close this PR and open a new one, feel free to ping me for review.

**Comment by dyudyunov** [comment]
@pomegranited could you close this PR in favor of https://github.com/openedx/edx-ora2/pull/2173?

**Comment by openedx-webhooks** [comment]
@Inferato Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-ora2&entry.867055334=https://github.com/openedx/edx-ora2/pull/2092&entry.1484655318=https://github.com/Inferato&entry.752974735=2023-10-25+15:20&entry.1917517419=2024-02-08+23:51&entry.2133058324=No) so we can improve your experience in the future.
